internal/config/config.go

```diff
@@ -128,14 +128,14 @@ func Get() *IngressConfig {
 			options.Set("KafkaUsername", *broker.Sasl.Username)
 			options.Set("KafkaPassword", *broker.Sasl.Password)
 			options.Set("SASLMechanism", *broker.Sasl.SaslMechanism)
-			options.Set("Protocol", *broker.Sasl.SecurityProtocol)
 		}
 		if broker.Cacert != nil {
 			caPath, err := cfg.KafkaCa(broker)
 			if err != nil {
 				panic("Kafka CA failed to write")
 			}
 			options.Set("KafkaCA", caPath)
+			options.Set("Protocol", *broker.SecurityProtocol)
```

Comment:
I think this should be moved outside of this if statement.  Based on the [librrdkafka config](https://github.com/confluentinc/librdkafka/blob/master/CONFIGURATION.md), this could be set to `plaintext`.  I read this as meaning that this setting could be set when not using a SSL/TLS/cacert.

Looking at the app-common-go code, it looks like SecurityProtocol is a pointer.  So we probably need to check for a nil.

I think we should default this setting to `plaintext` to match the default setting from librdkafka.